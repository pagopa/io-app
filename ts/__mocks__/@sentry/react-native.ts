export const captureException = jest.fn();
export const captureMessage = jest.fn();
export const captureEvent = jest.fn();
export const setContext = jest.fn();
export const setExtras = jest.fn();
export const setExtra = jest.fn();
export const setTags = jest.fn();
export const setTag = jest.fn();
export const setUser = jest.fn();
export const lastEventId = jest.fn();
export const captureCheckIn = jest.fn();
export const withMonitor = jest.fn();
export const flush = jest.fn().mockResolvedValue(false);
export const close = jest.fn().mockResolvedValue(false);
export const isInitialized = jest.fn().mockReturnValue(false);
export const isEnabled = jest.fn().mockReturnValue(false);
export const addEventProcessor = jest.fn();
export const startSession = jest.fn();
export const endSession = jest.fn();
export const captureSession = jest.fn();
export const ErrorBoundary = jest.fn(({ children }) => children);
