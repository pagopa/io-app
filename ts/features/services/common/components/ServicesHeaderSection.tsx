import React, { ComponentProps } from "react";
import { Dimensions, View } from "react-native";
import { useHeaderHeight } from "@react-navigation/elements";
import {
  ContentWrapper,
  IOColors,
  useIOTheme,
  VSpacer,
  WithTestID
} from "@pagopa/io-app-design-system";
import { ServicesHeader, ServicesHeaderSkeleton } from "./ServicesHeader";

const WINDOW_HEIGHT = Dimensions.get("window").height;

export type ServicesHeaderSectionProps = WithTestID<
  | ({
      isLoading?: false;
      extraBottomPadding?: number;
    } & ComponentProps<typeof ServicesHeader>)
  | {
      isLoading: true;
      extraBottomPadding?: number;
    }
>;

export const ServicesHeaderSection = ({
  extraBottomPadding,
  ...rest
}: ServicesHeaderSectionProps) => {
  const headerHeight = useHeaderHeight();
  const theme = useIOTheme();

  return (
    <View
      style={{
        backgroundColor: IOColors[theme["appBackground-secondary"]],
        paddingBottom: extraBottomPadding,
        paddingTop: WINDOW_HEIGHT + headerHeight,
        marginTop: -WINDOW_HEIGHT
      }}
    >
      <ContentWrapper>
        {rest.isLoading ? (
          <ServicesHeaderSkeleton />
        ) : (
          <ServicesHeader {...rest} />
        )}
        <VSpacer size={16} />
      </ContentWrapper>
    </View>
  );
};
