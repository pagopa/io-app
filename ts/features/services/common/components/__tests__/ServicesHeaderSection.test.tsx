import React from "react";
import { render } from "@testing-library/react-native";
import { ServicesHeaderSection } from "../ServicesHeaderSection";

jest.mock("@react-navigation/elements", () => ({
  useHeaderHeight: jest.fn().mockImplementation(() => 200)
}));

describe("ServicesHeaderSection component", () => {
  it("should match snapshot for loading", () => {
    const component = render(<ServicesHeaderSection isLoading={true} />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it("should match the snapshot for service header", () => {
    const component = render(
      <ServicesHeaderSection
        logoUri={require("../../../../../../img/test/logo.png")}
        subTitle={"#### subTitle ####"}
        title={"#### title ####"}
      />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
