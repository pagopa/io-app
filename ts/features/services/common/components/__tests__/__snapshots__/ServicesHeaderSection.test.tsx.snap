// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServicesHeaderSection component should match snapshot for loading 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#F4F5F8",
      },
      {
        "marginTop": -1334,
        "paddingBottom": undefined,
        "paddingTop": 1534,
      },
    ]
  }
>
  <View
    style={
      {
        "paddingHorizontal": 24,
      }
    }
  >
    <View
      accessible={true}
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingVertical": 6,
        }
      }
      testID="services-header-skeleton"
    >
      <View
        style={
          {
            "marginRight": 16,
          }
        }
      >
        <View
          collapsable={false}
          style={
            {
              "opacity": 0.5,
            }
          }
        >
          <View
            color="#D2D6E3"
            height={66}
            radius={12}
            style={
              {
                "backgroundColor": "#D2D6E3",
                "borderRadius": 12,
                "height": 66,
                "width": 66,
              }
            }
            width={66}
          />
        </View>
      </View>
      <View
        style={
          {
            "flex": 1,
          }
        }
      >
        <View
          collapsable={false}
          style={
            {
              "opacity": 0.5,
            }
          }
        >
          <View
            color="#D2D6E3"
            height={16}
            radius={8}
            style={
              {
                "backgroundColor": "#D2D6E3",
                "borderRadius": 8,
                "height": 16,
                "width": "100%",
              }
            }
            width="100%"
          />
        </View>
        <View
          style={
            {
              "height": 8,
            }
          }
        />
        <View
          collapsable={false}
          style={
            {
              "opacity": 0.5,
            }
          }
        >
          <View
            color="#D2D6E3"
            height={16}
            radius={8}
            style={
              {
                "backgroundColor": "#D2D6E3",
                "borderRadius": 8,
                "height": 16,
                "width": "80%",
              }
            }
            width="80%"
          />
        </View>
        <View
          style={
            {
              "height": 8,
            }
          }
        />
        <View
          collapsable={false}
          style={
            {
              "opacity": 0.5,
            }
          }
        >
          <View
            color="#D2D6E3"
            height={8}
            radius={8}
            style={
              {
                "backgroundColor": "#D2D6E3",
                "borderRadius": 8,
                "height": 8,
                "width": "60%",
              }
            }
            width="60%"
          />
        </View>
      </View>
    </View>
    <View
      style={
        {
          "height": 16,
        }
      }
    />
  </View>
</View>
`;

exports[`ServicesHeaderSection component should match the snapshot for service header 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#F4F5F8",
      },
      {
        "marginTop": -1334,
        "paddingBottom": undefined,
        "paddingTop": 1534,
      },
    ]
  }
>
  <View
    style={
      {
        "paddingHorizontal": 24,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingVertical": 6,
        }
      }
      testID="services-header"
    >
      <View
        style={
          {
            "marginRight": 16,
          }
        }
      >
        <View
          accessibilityIgnoresInvertColors={true}
          style={
            [
              {
                "borderColor": "rgba(14,15,19,0.1)",
                "borderCurve": "continuous",
                "borderWidth": 1,
                "overflow": "hidden",
              },
              {
                "backgroundColor": "#FFFFFF",
                "borderRadius": 12,
                "height": 66,
                "padding": 9,
                "width": 66,
              },
            ]
          }
        >
          <View
            style={
              [
                {
                  "backgroundColor": "#FFFFFF",
                  "borderCurve": "continuous",
                  "overflow": "hidden",
                },
                {
                  "borderRadius": 3,
                },
              ]
            }
          >
            <Image
              accessibilityIgnoresInvertColors={true}
              onError={[Function]}
              source={
                {
                  "testUri": "../../../img/test/logo.png",
                }
              }
              style={
                {
                  "height": "100%",
                  "resizeMode": "contain",
                  "width": "100%",
                }
              }
            />
          </View>
        </View>
      </View>
      <View
        style={
          {
            "flex": 1,
          }
        }
      >
        <Text
          allowFontScaling={false}
          dynamicTypeRamp="title2"
          maxFontSizeMultiplier={1.25}
          style={
            [
              {},
              {
                "color": "#2B2E38",
                "fontFamily": "Titillium Sans Pro",
                "fontSize": 22,
                "fontStyle": "normal",
                "fontWeight": "600",
                "lineHeight": 33,
              },
            ]
          }
        >
          #### title ####
        </Text>
        <Text
          allowFontScaling={false}
          dynamicTypeRamp="footnote"
          maxFontSizeMultiplier={1.25}
          style={
            [
              {},
              {
                "color": "#555C70",
                "fontFamily": "Titillium Sans Pro",
                "fontSize": 14,
                "fontStyle": "normal",
                "fontWeight": "700",
                "lineHeight": 21,
              },
            ]
          }
        >
          #### subTitle ####
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "height": 16,
        }
      }
    />
  </View>
</View>
`;
